### PR TITLE
Add google_gkeonprem_vmware_admin_cluster resource

### PR DESCRIPTION
**Pull request by naitianliu-google**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
Add `google_gkeonprem_vmware_admin_cluster` resource
```

### CODE REVIEW

**Comment by trodge** [comment]
The API does not seem to be returning the resource fields in the GET response after the create operation finishes.

The POST request looks like this:

```
POST /v1/projects/ci-test-project-188019/locations/us-west1/vmwareAdminClusters?alt=json&vmware_admin_cluster_id=basic71762183vq HTTP/1.1
Host: gkeonprem.googleapis.com
User-Agent: Terraform/1.8.3 (+https://www.terraform.io) Terraform-Plugin-SDK/2.33.0 terraform-provider-google-beta/acc
Content-Length: 800
Content-Type: application/json
Accept-Encoding: gzip

{
 "bootstrapClusterMembership": "projects/870316890899/locations/global/memberships/gkeonprem-terraform-test",
 "controlPlaneNode": {
  "cpus": 4,
  "memory": 8192,
  "replicas": 1
 },
 "description": "test admin cluster",
 "imageType": "ubuntu_containerd",
 "loadBalancer": {
  "f5Config": {
   "address": "10.251.135.22",
   "partition": "test-parition"
  },
  "vipConfig": {
   "addonsVip": "10.251.135.19",
   "controlPlaneVip": "10.251.133.5"
  }
 },
 "networkConfig": {
  "dhcpIpConfig": {
   "enabled": true
  },
  "podAddressCidrBlocks": [
   "192.168.0.0/16"
  ],
  "serviceAddressCidrBlocks": [
   "10.96.0.0/12"
  ]
 },
 "onPremVersion": "1.31.0-gke.35",
 "vcenter": {
  "address": "10.0.0.1",
  "caCertData": "test ca cert data",
  "cluster": "test cluster",
  "dataDisk": "test data disk",
  "datacenter": "test data center",
  "datastore": "test data store",
  "folder": "test folder",
  "resourcePool": "test resource pool"
 }
}
```

But the GET response looks like this:

```
{
  "name": "projects/ci-test-project-188019/locations/us-west1/vmwareAdminClusters/basic71762183vq",
  "uid": "3bcd26f7-f51e-4419-b5f1-501685f4c624",
  "etag": "kpSsF3gFDZWsQDA7ICinR-v2zx8AcClIpEj8KhwNPEk",
  "createTime": "2024-10-15T19:26:45.883897966Z",
  "updateTime": "2024-10-15T19:26:45.919645145Z"
}
```

Any idea why this would be the case?

File: mmv1/products/gkeonprem/VmwareAdminCluster.yaml [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+          controller creates a cluster or node pool. If the error message persists
+          for a longer period of time, it can be used to surface error message to
+          indicate real problems requiring user intervention.
+        output: true
+      - type: Array
+        name: 'conditions'
+        description: |
+          ResourceConditions provide a standard mechanism for higher-level status reporting from admin cluster controller.
+        output: true
+        item_type:
```
Comment (trodge) [inline mmv1/products/gkeonprem/VmwareAdminCluster.yaml:674]:
```suggestion
        item_type:
```
(remove trailing space)

**Comment by naitianliu-google** [comment]
@trodge Could you please help guide how to support empty message in terraform? We have a field `metal_lb_config` which only requires to provide an empty message, but it seems not supported in terraform. Do we have to add "enabled" subfield inside on API surface? Is there any workaround to bypass?

**Comment by trodge** [comment]
> @trodge Could you please help guide how to support empty message in terraform? We have a field `metal_lb_config` which only requires to provide an empty message, but it seems not supported in terraform. Do we have to add "enabled" subfield inside on API surface? Is there any workaround to bypass?

You could try using `send_empty_value`, but I'm not sure if it will work properly for a nested object. We would generally prefer to have an `enabled` subfield.

**Comment by naitianliu-google** [comment]
@trodge We have fixed the all issues mentioned in the PR. Could you please help take a look? Thanks!

**Review by trodge — APPROVED** [review]

**Pull request merged**